src/java/yaw/engine/SceneRenderer.java

```diff
@@ -101,14 +102,23 @@ public void render(Camera pCamera, ShaderManager shaderManager) {
 
         for (Mesh mesh : mMeshMap.keySet()) {
             ShaderProperties meshProps = mesh.getShaderProperties(lightModel);
-            // TODO : ugly cast, fix when support for e.g. PBR materials
```

Comment:
At some point yes, the shaderProgram should be instantiated properly... Scenes should support both ADS & PBR materials